airgun/views/common.py

```diff
@@ -383,6 +386,71 @@ class add_tab(AddTab):
         )
 
 
+class NewAddRemoveResourcesView(View):
+    searchbox = PF4Search()
+    type = Dropdown(
+        locator='.//div[contains(@class, "All repositories") or'
+        ' contains(@aria-haspopup="listbox")]'
+    )
+    Status = Dropdown(
+        locator='.//div[contains(@class, "All") or contains(@aria-haspopup="listbox")]'
+    )
+    add_repo = PF4Button('OUIA-Generated-Button-secondary-2')
+    # Need to add kebab menu
+    table = PatternflyTable(
+        component_id='OUIA-Generated-Table-4',
+        column_widgets={
+            0: Checkbox(locator='.//input[@type="checkbox"]'),
+            'Type': Text('.//a'),
+            'Name': Text('.//a'),
+            'Product': Text('.//a'),
+            'Sync State': Text('.//a'),
+            'Content': Text('.//a'),
+            'Status': Text('.//a'),
+        },
+    )
+
+    def search(self, value):
+        """Search for specific available resource and return the results"""
+        self.searchbox.search(value)
+        # Tried following ways to wait for table to be displayed, only sleep worked
+        # Might need a before/after fill
+        wait_for(
+            lambda: self.table.is_displayed is True,
+            timeout=60,
+            delay=1,
+        )
+        time.sleep(3)
```

Comment:
avoid the hard sleep

Reply:
That's what the comment on line 416 is about. He tried. Sam, what have you tried? Ensure_page_safe, wait_displayed?

Reply:
I can investigate. A lot of this stuff is from an older PR I salvaged to begin work on CV UI component eval, and I didn't clean it up properly.